data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/cnec/CsaProfileFlowCnecCreator.java

```diff
@@ -0,0 +1,390 @@
+            csaProfileFlowCnecCreationContexts.add(CsaProfileFlowCnecCreationContext.notImported(assessedElementId, ImportStatus.NOT_FOR_RAO, "AssessedElementWithContingency.combinationConstraintKind is considered"));
+            return null;
+        }
+        if (CsaProfileConstants.ASSESSED_ELEMENT_WITH_CONTINGENCIES_LINK_INCLUDED.equals(combinationConstraintKind) && !isCombinableWithContingency) {
+            Contingency contingencyToLink = crac.getContingencies().stream().filter(contingency -> contingency.getId().equals(contingencyId)).findFirst().orElse(null);
+            if (contingencyToLink == null) {
+                csaProfileFlowCnecCreationContexts.add(CsaProfileFlowCnecCreationContext.notImported(assessedElementId, ImportStatus.INCONSISTENCY_IN_DATA, "the contingency " + contingencyId + " linked to the assessed element doesn't exist in the CRAC"));
+                return null;
+            }
+            combinableContingencies.add(contingencyToLink);
+        } else if (CsaProfileConstants.ASSESSED_ELEMENT_WITH_CONTINGENCIES_LINK_EXCLUDED.equals(combinationConstraintKind) && isCombinableWithContingency) {
+            Set<Contingency> combinableContingenciesAfterFilter =
+                    combinableContingencies.stream().filter(contingency -> !contingency.getId().equals(contingencyId)).collect(Collectors.toSet());
+            if (combinableContingenciesAfterFilter.size() == combinableContingencies.size()) {
+                csaProfileFlowCnecCreationContexts.add(CsaProfileFlowCnecCreationContext.notImported(assessedElementId, ImportStatus.INCONSISTENCY_IN_DATA, "the contingency " + contingencyId + " excluded from the contingencies linked to the assessed element doesn't exist in the CRAC"));
+                return null;
+            }
+            combinableContingencies = combinableContingenciesAfterFilter;
+        } else {
+            csaProfileFlowCnecCreationContexts.add(CsaProfileFlowCnecCreationContext.notImported(assessedElementId, ImportStatus.INCONSISTENCY_IN_DATA, "AssessedElementWithContingency.combinationConstraintKind and AssessedElement.isCombinableWithContingency have inconsistent values"));
+            return null;
+        }
+        return combinableContingencies;
+    }
+
+    private boolean addCurrentLimit(String assessedElementId, FlowCnecAdder flowCnecAdder, PropertyBag currentLimit, Boolean isCombinableWithContingency) {
+        String currentLimitId = currentLimit.getId(CsaProfileConstants.REQUEST_CURRENT_LIMIT_OPERATIONAL_LIMIT_TERMINAL);
+        Identifiable<?> networkElement = this.getNetworkElementInNetwork(currentLimitId);
+        if (networkElement == null) {
+            csaProfileFlowCnecCreationContexts.add(CsaProfileFlowCnecCreationContext.notImported(assessedElementId, ImportStatus.INCONSISTENCY_IN_DATA, "current limit equipment is missing in network : " + currentLimitId));
```

Comment:
ImportStatus.Element not found in network

Reply:
done